src/command/dns-command.ts

```diff
@@ -0,0 +1,64 @@
+import {getServers as getDnsServers} from 'node:dns';
+import Joi from 'joi';
+import type {Socket} from 'socket.io-client';
+import dig, {DnsQueryResult} from 'node-dig-dns';
+import type {CommandInterface} from '../types.js';
+import {InvalidOptionsException} from './exception/invalid-options-exception.js';
+
+type DnsOptions = {
+	type: 'dns';
+	target: string;
+	query: {
+		type?: string[];
+		address?: string;
+		protocol?: string;
+		port?: number;
+	};
+};
+
+const allowedTypes = ['A', 'AAAA', 'ANY', 'CNAME', 'DNSKEY', 'DS', 'MX', 'NS', 'NSEC', 'PTR', 'RRSIG', 'SOA', 'TXT', 'SRV'];
+
+const dnsOptionsSchema = Joi.object<DnsOptions>({
+	type: Joi.string().valid('dns'),
+	target: Joi.string(),
+	query: Joi.object({
+		type: Joi.string().valid(...allowedTypes).optional(),
+		address: Joi.string().optional(),
+		protocol: Joi.string().valid().optional(),
+		port: Joi.number().optional(),
+	}),
+});
+
+export const dnsCmd = async (options: DnsOptions): Promise<DnsQueryResult> => {
+	const protocolArg = options.query.protocol?.toLowerCase() === 'tcp' ? '+tcp' : '';
+
+	const args = [
+		options.target,
+		`@${options.query.address ?? getDnsServers().pop()!}`,
```

Comment:
This is an optimistic approach 🙂 . Dns servers returned by `dns.getServers()` may include addresses with a port as well as ipv6 addresses. Why not just avoid the `@servername` option if no `options.query.address` provided?

Reply:
Maybe. That's a leftover from the previous library - it wouldn't find its own defaults.

Reply:
We should let dig use the system defaults for now, so dont provide anything manually